### PR TITLE
fix tito builder now that provides exceeds argument restriction on subprocess

### PR DESCRIPTION
**Pull request by maxamillion**
Fix for https://github.com/openshift/origin/issues/11125

Signed-off-by: Adam Miller <maxamillion@fedoraproject.org>

### CODE REVIEW

File: .tito/lib/origin/builder/__init__.py [hunk 1]
```diff
@@ -105,12 +105,24 @@ def _setup_test_specfile(self):
...
+            # count for what we can pass into sed via subprocess because there
+            # are so many bundled libraries.
+            with open(self.spec_file, 'r') as spec_file_f:
+                spec_file_lines = spec_file_f.readlines()
+            with open(self.spec_file, 'w') as spec_file_f:
+                for line in spec_file_lines:
+                    if '### AUTO-BUNDLED-GEN-ENTRY-POINT' in line:
+                        for provides in provides_list:
+                            # Because $reasons
+                            if "'" in provides:
```
Comment (stevekuznetsov) [inline .tito/lib/origin/builder/__init__.py]: The `replace` will be a no-op if this is false, so I think we could use:
```py
if '### AUTO-BUNDLED-GEN-ENTRY-POINT' in line:
    spec_file_f.write('\n'.join([provides.replace('"', '').replace("'", '') for provides in provides_list]))
```

Comment (maxamillion) [inline .tito/lib/origin/builder/__init__.py]: That's fine, I don't have a real preference on this. Will fix.

**Comment by stevekuznetsov** [comment]
Changes LGTM, worked fine in tests. Can merge on squash and OK from @sdodson

**Comment by sdodson** [comment]
Changes LGTM, gave it a quick test and looks good lets squash though.

**Comment by maxamillion** [comment]
squashed! :)

**Comment by stevekuznetsov** [comment]
[merge]

**Pull request merged**